harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -751,14 +751,14 @@ def _train_for_op(
         self, op: core.SearcherOperation, train_boundaries: List[_TrainBoundary]
     ) -> None:
         if self.local_training:
-            searcher_length = self.max_length
+            train_length = Batch(1) if self.test_mode else self.max_length
         else:
-            searcher_length = TrainUnit._from_searcher_unit(
+            train_length = TrainUnit._from_searcher_unit(
                 op.length, self.searcher_unit, self.global_batch_size
             )
-        assert searcher_length
+        assert train_length
```

Comment:
just confirming, we don't support `--test` without `--local` through any codepath except the legacy codepath, right?

This seems correct under that condition, but might make more sense to check for test mode first, then local, since if we did support nonlocal test mode we would still only train one batch.

Reply:
good point. updated.